nixos/modules/services/networking/amuled.nix

```diff
@@ -66,15 +68,37 @@ in
       wantedBy = [ "multi-user.target" ];
       after = [ "network.target" ];
 
-      preStart = ''
-        mkdir -p ${cfg.dataDir}
-        chown ${user} ${cfg.dataDir}
-      '';
-
-      script = ''
-        ${pkgs.su}/bin/su -s ${pkgs.runtimeShell} ${user} \
-            -c 'HOME="${cfg.dataDir}" ${pkgs.amule-daemon}/bin/amuled'
-      '';
+      serviceConfig = {
+        Environment = [ "HOME=${cfg.dataDir}" ];
```

Comment:
Maybe use `systemd.services.amuled.environment.HOME` instead?

Reply:
What's the difference between the two?

Reply:
`systemd.services.amuled.environment` takes an attrset, which would make it easier to override values using the module system.

Reply:
Fair enough, added in https://github.com/NixOS/nixpkgs/pull/354517/commits/a11b428ba88efea016031b468872a91fb93b0e1c, thanks!